## This script is used for expression-level dominant (ELD) and transgressive expression analysis in long-liguled T. miscellus (Tml)

In [1]:
import pandas as pd
import os

In [2]:
os.chdir("/ufrc/soltis/shan158538/TragFL_NewAnalysis/OutPut/DE_analysis/Additive_expression_Tml/")

### 1. Transgressive expression analysis

In [3]:
## Input files
nonAdditive = pd.read_table("DESeq2_Tml_nonAdditive_loci.txt", sep = "\t")
DE_Tml_higher_than_Tdu = pd.read_table("DESeq2_DE_loci_Tml_higher_than_Tdu.txt", sep = "\t")
DE_Tml_higher_than_Tpr = pd.read_table("DESeq2_DE_loci_Tml_higher_than_Tpr.txt", sep = "\t")

DE_Tdu_higher_than_Tml = pd.read_table("DESeq2_DE_loci_Tdu_higher_than_Tml.txt", sep = "\t")
DE_Tpr_higher_than_Tml = pd.read_table("DESeq2_DE_loci_Tpr_higher_than_Tml.txt", sep = "\t")

In [4]:
nonAdditive.head()

baseMean    ...         padj
Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c3_g1  103.883699    ...     0.028670
Tpr_TRINITY_DN12798_c2_g2|Tdu_TRINITY_DN24179_c...   53.687590    ...     0.029052
Tpr_TRINITY_DN11712_c1_g8|Tdu_TRINITY_DN19492_c...  439.879994    ...     0.000084
Tpr_TRINITY_DN14614_c1_g6|Tdu_TRINITY_DN1401_c0_g1   71.441805    ...     0.012264
Tpr_TRINITY_DN13868_c2_g1|Tdu_TRINITY_DN24297_c...   16.868174    ...     0.035224

[5 rows x 6 columns]

In [5]:
## Intersections
## Tml higher than both Tdu and Tpr
nonAdditive_Tml_higher_than_Tdu = pd.merge(nonAdditive, 
                                           DE_Tml_higher_than_Tdu, how='inner', 
                                           left_index=True, right_index=True)
nonAdditive_Tml_higher_than_Tdu_and_Tpr = pd.merge(nonAdditive_Tml_higher_than_Tdu, 
                                                   DE_Tml_higher_than_Tpr, 
                                                   how='inner', left_index=True, right_index=True)

In [6]:
nonAdditive_Tml_higher_than_Tdu_and_Tpr.shape

(180, 18)

In [7]:
## Intersections
## Tml lower than both Tdu and Tpr
nonAdditive_Tml_lower_than_Tdu = pd.merge(nonAdditive, 
                                          DE_Tdu_higher_than_Tml, how='inner', 
                                          left_index=True, right_index=True)
nonAdditive_Tml_lower_than_Tdu_and_Tpr = pd.merge(nonAdditive_Tml_lower_than_Tdu, 
                                                  DE_Tpr_higher_than_Tml, 
                                                  how='inner', left_index=True, right_index=True)

In [8]:
nonAdditive_Tml_lower_than_Tdu_and_Tpr.shape

(247, 18)

In [9]:
## Write out
nonAdditive_Tml_higher_than_Tdu_and_Tpr.to_csv("Tml_transgressive_higher.txt", columns=[], header=False)
nonAdditive_Tml_lower_than_Tdu_and_Tpr.to_csv("Tml_transgressive_lower.txt", columns=[], header=False)

**180** loci are transgressively higher expressed in Tml

**247** loci are transgressively lower expressed in Tml

### 2. Expression-level dominant

In [10]:
## Inputfiles
DE_Tdu_higher_than_Tpr = pd.read_table("DESeq2_DE_loci_Tdu_higher.txt", sep = "\t")
DE_Tpr_higher_than_Tdu = pd.read_table("DESeq2_DE_loci_Tpr_higher.txt", sep = "\t")
Tml_same_as_Tdu = pd.read_table("DESeq2_noneDE_loci_Tml_Tdu.txt", sep = "\t")
Tml_same_as_Tpr = pd.read_table("DESeq2_noneDE_loci_Tml_Tpr.txt", sep = "\t")

In [11]:
## Intersections
## Expression-level dominant of Tdu
nonAdditive_ELDofTdu = pd.merge(nonAdditive, Tml_same_as_Tdu, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTdu_TduHigher = pd.merge(nonAdditive_ELDofTdu, DE_Tdu_higher_than_Tpr, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTdu_TprHigher = pd.merge(nonAdditive_ELDofTdu, DE_Tpr_higher_than_Tdu, how='inner', left_index=True, right_index=True)

In [12]:
nonAdditive_ELDofTdu_TduHigher.shape

(62, 18)

In [13]:
nonAdditive_ELDofTdu_TprHigher.shape

(45, 18)

In [14]:
## Intersections
## Expression-level dominant of Tpr
nonAdditive_ELDofTpr = pd.merge(nonAdditive, Tml_same_as_Tpr, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTpr_TduHigher = pd.merge(nonAdditive_ELDofTpr, DE_Tdu_higher_than_Tpr, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTpr_TprHigher = pd.merge(nonAdditive_ELDofTpr, DE_Tpr_higher_than_Tdu, how='inner', left_index=True, right_index=True)

In [15]:
nonAdditive_ELDofTpr_TduHigher.shape

(58, 18)

In [16]:
nonAdditive_ELDofTpr_TprHigher.shape

(61, 18)

In [17]:
## Write out
nonAdditive_ELDofTdu_TduHigher.to_csv("Tml_ELDofTdu_TduHigher.txt", columns=[], header=False)
nonAdditive_ELDofTdu_TprHigher.to_csv("Tml_ELDofTdu_TprHigher.txt", columns=[], header=False)

nonAdditive_ELDofTpr_TduHigher.to_csv("Tml_ELDofTpr_TduHigher.txt", columns=[], header=False)
nonAdditive_ELDofTpr_TprHigher.to_csv("Tml_ELDofTpr_TprHigher.txt", columns=[], header=False)

**62** loci showed ELD of Tdu with higher expression in Tdu than Tpr

**45** loci showed ELD of Tdu with higher expression in Tpr than Tdu

**58** loci showed ELD of Tpr with higher expression in Tdu than Tpr

**61** loci showed ELD of Tpr with higher expression in Tpr than Tdu